sky/packages/sky/lib/sky_tool

```diff
@@ -251,10 +224,171 @@ class StartSky(object):
                     app_path = os.path.join(args.sky_src_path, args.ios_sim_debug_build_path, IOS_APP_NAME)
                 IOSSimulator.fork_install_app(app_path)
 
+        # TODO(iansf): fix this so that we don't have to pass sky_server_root, main_dart, pid, and args.
+        android.setup_servers(sky_server_root, main_dart, pids, args)
+
+
+class StopSky(object):
+    def add_subparser(self, subparsers):
+        stop_parser = subparsers.add_parser('stop',
+            help=('kill all running SkyShell.apk processes'))
+        stop_parser.set_defaults(func=self.run)
+
+    def _run(self, args):
+        with open('/dev/null', 'w') as dev_null:
+            logging.info(' '.join(args))
+            subprocess.call(args, stdout=dev_null, stderr=dev_null)
+
+    def run(self, args, pids):
+        self._run(['fuser', '-k', '%s/tcp' % SKY_SERVER_PORT])
+
+        if 'remote_sky_server_port' in pids:
+            port_string = 'tcp:%s' % pids['remote_sky_server_port']
+            self._run([AndroidDevice().adb_path, 'reverse', '--remove', port_string])
+
+        self._run([AndroidDevice().adb_path, 'shell', 'am', 'force-stop', ANDROID_PACKAGE])
+
+        pids.clear()
+
+class AndroidDevice(object):
```

Comment:
Chrome has its own version of this, which I've previously eschewed because it was so complicated.
https://code.google.com/p/chromium/codesearch#chromium/src/build/android/pylib/